pycontrails/core/models.py

```diff
@@ -731,25 +762,271 @@ def interpolate_met(
     KeyError
         Parameter ``met_key`` not found in ``met``.
     """
-    if vector_key is None:
-        vector_key = met_key
+    vector_key = vector_key or met_key
 
     if (out := vector.get(vector_key, None)) is not None:
         return out
 
     if met is None:
-        raise ValueError(f"No variable key {vector_key} in ``vector`` and ``met`` is None")
+        raise ValueError(f"No variable key '{vector_key}' in 'vector' and 'met' is None")
+
+    if met_key in ("q", "specific_humidity"):
+        mda, log_applied = _extract_q(met, met_key, q_method)
+        out = interpolate_gridded_specific_humidity(
+            mda, vector, q_method, log_applied, **interp_kwargs
+        )
+
+    else:
+        try:
+            mda = met[met_key]
+        except KeyError as exc:
+            raise KeyError(f"No variable key '{met_key}' in 'met'.") from exc
+
+        out = vector.intersect_met(mda, **interp_kwargs)
+
+    vector[vector_key] = out
+    return out
+
+
+def _extract_q(met: MetDataset, met_key: str, q_method: str) -> tuple[MetDataArray, bool]:
```

Comment:
I think you could move everything below to the `core.interpolation` module to make it clear.